std/typecons.d

```diff
@@ -80,40 +79,69 @@ public:
     ---
     */
     static Unique!T create(A...)(auto ref A args)
-    if (__traits(compiles, new T(args)))
     {
+        import core.memory : GC;
+        import core.stdc.stdlib : malloc;
+        import std.conv : emplace;
+        import std.exception : enforce;
+
         debug(Unique) writeln("Unique.create for ", T.stringof);
         Unique!T u;
-        u._p = new T(args);
+
+        // TODO: May need to fix alignment?
+        // Does emplace still need to mess with alignment if
+        // the memory is coming from malloc, or does malloc handle that?
+
+        static if (is(T == class))
+            immutable size_t allocSize = __traits(classInstanceSize, T);
+        else
+            immutable size_t allocSize = T.sizeof;
+
+        void* rawMemory = enforce(malloc(allocSize), "malloc returned null");
```

Comment:
Thanks! For now I'll emulate what you do in #3031, and once both are hopefully merged, we can move over to that. Hopefully this will also fix the segfault I mentioned above.